### PR TITLE
Add tests to the search product feature

### LINKED ISSUE #12145: [BUU] Improve Product Search Keywords
### What we should change and why (this is tech debt)

We have a test for this feature in which we assert only if the data is saved on the respective fields. We should improve it by testing the search feature on the products page, making sure it finds products according to the added search keywords.

Relevant spec:
https://github.com/openfoodfoundation/openfoodnetwork/blob/78a1726bdc3bf61194a9e68c112865da2df3c5b6/spec/system/admin/products_spec.rb#L494

### Context

This came up here:
https://openfoodnetwork.slack.com/archives/C01CXQNJ1J6/p1707424388711889

### PR DESCRIPTION
**Pull request by cyrillefr**
#### What? Why?

- Closes #12145 

There a products search feature that can be used by customers and we don't know how it works.
I have searched how the feature works and added matching tests.

The search is done in `app/controller/api/vo/order_cycles_controller.rb` in the `products` method.
More precisely, it done by the ransack gem, with a search parameter that looks like this:
`"name_or_meta_keywords_or_variants_display_as_or_variants_display_name_or_supplier_name_cont"`

That implies the search is done against the name or keywords from products; variant_display or variant_display_name from product variants; supplier name of the product.
The "_cont" suffix means this not a exact matching search but rather the search term should be not only exactly present in DB but also only a part of it can be matched. 

#### What should we test?

It is an addition of examples in automated specs to illustrate the possibilities of the actual search feature, so there should be no manual testing required here since the back-end code is untouched.

But for the record, the search feature can be found:
- connect as a customer
- chose a supplier and begin shopping
- you should be able to search (and that what the tests show) for:
  - name of a product
  - name of a supplier
  - names of variants attached to a product
  - some keywords that have previously been entered in the back-office

**NB** The search feature in the back-office is not the same, is not implemented in the same place and therefore do not yield the same results.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [X] Technical changes only
- [ ] Feature toggled

### CODE REVIEW

**Comment by cyrillefr** [comment]
Hello @filipefurtad0 , found out a bit too late that I have messed up a bit with titles of my tests. I will change them with the requested changes if any.

File: spec/system/consumer/shopping/shopping_spec.rb [hunk 1]
```diff
@@ -330,24 +333,52 @@
...
+            expect(page).to have_content product.variants[2].display_name # not Badgers
+          end
+        end
+        it 'returns results by searching by variants display_name' do
+          # model: Enterprise name
+          fill_in "search", with: "Enterp" # Enterprise 1 sells nothing
+          within('p.no-results') do
+            expect(page).to have_content "Sorry, no results found for Enterp"
+          end
+        end
```
Comment (mkllnk) [inline spec/system/consumer/shopping/shopping_spec.rb:381]: Can we combine these `it` blocks? It's good practice to keep them separate as you did but it's very slow in reality and consumes unnecessary energy on every test run. CI can run 100 times a day.

**Review by mkllnk — APPROVED** [review]

**Review by dacook — CHANGES_REQUESTED** [review]
Looks good thanks, but as you and Maikel both suggested changing the examples arrangement, I'll request changes (with Maikel's suggestion preferred).

File: spec/system/consumer/shopping/shopping_spec.rb [hunk 2]
```diff
@@ -330,24 +333,52 @@
...
+          # model: meta_keywords
+          fill_in "search", with: "Wild" # For product named "Meercats"
+          expect(page).to have_content product2.name
+        end
+        it 'returns results by searching by variants display_name' do
+          # model: variant display_name
+          fill_in "search", with: "Ferrets" # For variants named "Ferrets"
+          within('div.pad-top') do
+            expect(page).to have_content product2.variants[1].display_name # Ferrets
+            expect(page).not_to have_content product.variants[2].display_name # not Badgers
```
[the remark this answers is not in the document]
Comment (cyrillefr) [reply spec/system/consumer/shopping/shopping_spec.rb]: I have made changes in that way.

[the remark this answers is not in the document]
Comment (cyrillefr) [reply spec/system/consumer/shopping/shopping_spec.rb]: I have trimmed down to 2 examples ( to avoid one too big one). It is faster.

**Review by mkllnk — APPROVED** [review]
Thank you.

**Pull request merged**